File: solution/1992. Find All Groups of Farmland/1992.cpp
class Solution {
public:
    vector<vector<int>> findFarmland(vector<vector<int>>& land) {
        const int m = land.size(), n = land[0].size();
        
        vector<vector<int>> res;
        for (int i = 0; i < m; ++i) {
            for (int j = 0; j < n; ++j) {
                if (land[i][j] != 1) continue;
                int y = i, x = j;
                for (; y < m && land[y][j] == 1; ++y);
                for (; x < n && land[i][x] == 1; ++x);
                --y;
                --x;
                res.push_back({i, j, y, x});
                
                for (int a = i; a <= y; ++a)
                    for (int b = j; b <= x; ++b)
                        land[a][b] = 2;
            }
        }
        return res;
    }
};

File: solution/1992. Find All Groups of Farmland/1992.py
class Solution:
    def findFarmland(self, land: List[List[int]]) -> List[List[int]]:
        m, n = len(land), len(land[0])
        
        res = []
        for i in range(m):
            for j in range(n):
                if land[i][j] != 1:
                    continue
                y, x = i, j
                while y < m and land[y][j] == 1:
                    y += 1
                while x < n and land[i][x] == 1:
                    x += 1
                y -= 1
                x -= 1
                res.append([i, j, y, x])
                
                for a in range(i, y + 1):
                    for b in range(j, x + 1):
                        land[a][b] = 2
        return res
                    
        
